DataOnion/Extensions.cs

```diff
@@ -1,47 +1,85 @@
 using System.Data.Common;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.DependencyInjection;
-using DataOnion.Config;
 using DataOnion.db;
 
 namespace DataOnion
 {
-    public static class DataHelperExtensions
+    public interface IFluentDatabaseOnion : IFluentDapperSetup, IFluentEfCoreSetup {}
+
+    public interface IFluentDapperSetup
     {
-        public static IServiceCollection ConfigureEfCoreOnion<TDbContext>(
-            this IServiceCollection serviceCollection,
-            EFCoreOptions? options = null
+        IFluentEfCoreSetup ConfigureEfCore<T>(
+            Func<string, Action<DbContextOptionsBuilder>>? dataConnector,
+            ServiceLifetime contextLifetime = ServiceLifetime.Scoped,
+            ServiceLifetime optionsLifetime = ServiceLifetime.Scoped
+        ) where T : DbContext;
+    }
+
+    public interface IFluentEfCoreSetup
+    {
+        IFluentDapperSetup ConfigureDapper<T>(
+            Func<string, T> connectionGetter
+        ) where T : DbConnection;
+    }
+
+    public class FluentDatabaseOnion : IFluentDatabaseOnion
+    {
+        private readonly IServiceCollection _serviceCollection;
+        private readonly string _connectionString;
+
+        public FluentDatabaseOnion(
+            IServiceCollection serviceCollection,
+            string connectionString
         )
-            where TDbContext : DbContext
         {
-            serviceCollection.AddDbContext<TDbContext>(
-                options?.DataConnector?.Invoke(options.ConnectionString),
-                options?.ServiceLifetime ?? ServiceLifetime.Scoped,
-                options?.OptionsLifetime ?? ServiceLifetime.Scoped
+            _serviceCollection = serviceCollection;
+            _connectionString = connectionString;
+        }
+
+        public IFluentDapperSetup ConfigureDapper<T>(
+            Func<string, T> connectionGetter
+        ) where T: DbConnection
+        {
+            _serviceCollection.AddScoped<T>(
+                services => connectionGetter(_connectionString)
+            );
+
+            return this;
```

Comment:
I believe I understand whats happening but just incase I'm wrong. 

This is using method chaining to setup EfCore and Dapper services